elfutils: Error linking libraries against `elfutils` with `undefined symbol: eu_search_tree_init`

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

Statically linking an application against `elfutils` fails with `error: undefined symbol: eu_search_tree_init`. 

### Steps to reproduce

This can be easily reproduced in my application. Apologies that it's not, erm, minimal:

- `git clone https://github.com/javierhonduco/lightswitch`
- `cd lightswitch && nix flake update`
- `nix build`

This seems to be this bug reported upstream https://sourceware.org/bugzilla/show_bug.cgi?id=32293. I don't think a new version that has this bugfix as been released.

### Expected behaviour

Linking should succeed

### Screenshots

_No response_

### Relevant log output

```console
>   = note: rust-lld: error: undefined symbol: eu_search_tree_fini
       >           >>> referenced by elf_end.o:(elf_end) in archive /build/nzfk4l14s8c77zjgxyywy81giryzs6mh-source/target/release/deps/liblibbpf_sys-f6e8a042e99e77a3.rlib
       >           
       >           rust-lld: error: undefined symbol: eu_search_tree_init
       >           >>> referenced by elf_begin.o:(file_read_elf) in archive /build/nzfk4l14s8c77zjgxyywy81giryzs6mh-source/target/release/deps/liblibbpf_sys-f6e8a042e99e77a3.rlib
       >           >>> referenced by elf_begin.o:(file_read_elf) in archive /build/nzfk4l14s8c77zjgxyywy81giryzs6mh-source/target/release/deps/liblibbpf_sys-f6e8a042e99e77a3.rlib
       >           collect2: error: ld returned 1 exit status
       >           
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.7-200.fc40.x86_64, Fedora Linux, 40 (Workstation Edition), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/qpjsygnxp61armwi0rhk3yw7wygc27ch-source`


### Notify maintainers

@r-burns 

### I assert that this issue is relevant for Nixpkgs

- [X] I assert that this is a bug and not a support request.
- [X] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [X] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/el/elfutils/package.nix
+++ b/pkgs/by-name/el/elfutils/package.nix
@@ -29,11 +29,11 @@
 # TODO: Look at the hardcoded paths to kernel, modules etc.
 stdenv.mkDerivation rec {
   pname = "elfutils";
-  version = "0.192";
+  version = "0.193";
 
   src = fetchurl {
     url = "https://sourceware.org/elfutils/ftp/${version}/${pname}-${version}.tar.bz2";
-    hash = "sha256-YWCZvq4kq6Efm2PYbKbMjVZtlouAI5EzTJHfVOq0FrQ=";
+    hash = "sha256-eFf0S2JPTY1CHfhRqq57FALP5rzdLYBJ8V/AfT3edjU=";
   };
 
   patches =
@@ -145,6 +145,14 @@ stdenv.mkDerivation rec {
     # can be executed, so we need to match build and host platform exactly.
     && (stdenv.hostPlatform == stdenv.buildPlatform);
   doInstallCheck = !stdenv.hostPlatform.isMusl && (stdenv.hostPlatform == stdenv.buildPlatform);
+
+  preCheck = ''
+    # Workaround lack of rpath linking:
+    #   ./dwarf_srclang_check: error while loading shared libraries:
+    #     libelf.so.1: cannot open shared object file: No such file or directory
+    # Remove once https://sourceware.org/PR32929 is fixed.
+    export LD_LIBRARY_PATH="$PWD/libelf:$LD_LIBRARY_PATH"
+  '';
 
   passthru.updateScript = gitUpdater {
     url = "https://sourceware.org/git/elfutils.git";

